src/main/java/edu/nextstep/racingcar/step3/app/Car.java

```diff
@@ -0,0 +1,31 @@
+package edu.nextstep.racingcar.step3.app;
+
+import java.util.function.Supplier;
+
+public class Car implements Vehicle {
+
+    private final Supplier<Boolean> movement;
+    private int distance = 1;
+
+    public Car(ParamsValidator paramsValidator, Supplier<Boolean> movement) {
+        paramsValidator.validate();
+
+        this.movement = movement;
+    }
+
+    public boolean isMove() {
+        return movement.get();
+    }
+
+    public void setDistance(int distance) {
+        this.distance = distance;
+    }
```

Comment:
현실세계를 잘 생각해보면 setDistance는 움직이는 자동차의 위치를 갑자기 순간이동 시키는것과 같다고 생각해요.
메소드 이름만 놓고 보면 그렇네요 ㅎㅎ
그래서 distance를 변경하기보단 한다면 그냥 생성자를 통해서 새로운 Car를 만들어주는게 좋지 않을까요? 이 때 distance를 넘겨주는게 조금 더 객체지향 스러울 것 같습니다.
```suggestion
    public Car(int distance) {
        this.distance = distance;
    }
```